Library/Homebrew/formula_installer.rb

```diff
@@ -1001,14 +1001,48 @@ def link(keg)
     @show_summary_heading = true
   end
 
+  sig { params(service_hash: Hash).String }
+  def generate_plist(service_hash)
+    items = ["<key>Label</key><string>#{formula.plist_name}</string>"]
+    items << "<key>KeepAlive</key><true/>" if service_hash[:keep_alive]
+    items << "<key>RunAtLoad</key><true/>" if service_hash[:run_at_load]
+    items << "<key>ProgramArguments</key>"
+    items << "<array>"
+    service_hash[:arguments].each do |item|
+      items << "<string>#{item}</string>"
+    end
+    items << "</array>"
+    items << "<key>WorkingDirectory</key><string>#{service_hash[:working_dir]}</string>" if service_hash[:working_dir]
+    items << "<key>StandardOutPath</key><string>#{service_hash[:log_path]}</string>" if service_hash[:log_path]
+    items << "<key>StandardErrorPath</key><string>#{service_hash[:error_path]}</string>" if service_hash[:error_path]
+
+    <<~EOS
+      <?xml version="1.0" encoding="UTF-8"?>
+      <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+      <plist version="1.0">
+        <dict>
+        #{items.join("\n")}
+        </dict>
+      </plist>
+    EOS
+  end
+
   sig { void }
   def install_plist
```

Comment:
```suggestion
  def install_service
```

given it's a private API anyway?